src/CustomElementInternals.js

```diff
@@ -208,6 +222,11 @@ export default class CustomElementInternals {
             this.patchAndUpgradeTree(importNode, visitedImports);
           });
         }
+      } else if (element.hasAttribute('is') || element.is) {
+        element.__CE_is = element.getAttribute('is') || element.is;
+        element.removeAttribute('is');
+        delete element.is;
+        elements.push(element);
```

Comment:
You could limit the calls to `hasAttribute, getAttribute, removeAttribute` to have better perf. Also, avoid `delete element.is` as it's slow and all we need here is to unset it.

e.g.
```js
} else {
  // if getAttribute returns an empty string consider it null.
  const is = element.is || element.getAttribute('is') || null;
  if (is) {
    element.__CE_is = is;
    element.is = null;
    element.removeAttribute('is');
  }
  elements.push(element);
}